WebClient/doc/components/temperatureControl/temperatureControl.md

```diff
@@ -0,0 +1,84 @@
+# TemperatureControl
+
+Control that can be used to either view the live temperature condition or set a temperature value to see its effects.
+
+![Control structure design](images/temperatureControl.PNG) 
+
+## Features
+
+* Displays live temperature data (by default).
+* Temperature can be changed by the following 3 ways:
+    * Clicking anywhere inside the thermometer.
+    * Clicking on the mercury and dragging it up or down.
+    * Change the temperature value in the input (textbox) next to the thermometer.
+* Following temperature units are supported - K, °C, °F, °R
+* Click on the label next to the thermometer to view a dropdown displaying all the available units.
+
+## Architecture
+
+![Class Diagram](images/temperatureControl_classDiagram.png)
+
+This component consists of a Model-View design.
+
+**temperatureControlModel** is the source of the data. It contains 2 parameters along with their getters and setters - value & temperatureUnit. This model supports a subscription mechanism for  tracking its state. Whenever the model state is changed, a corresponding event is emitted.
+
+**temperatureControlView** visually represents the state of the model using a thermometer. By default, the live values for the temperature are displayed but, when a user makes changes to the temperature value, a call is made to the model to update the parameters. This leads to an event change event which leads to the visual components being updated to reflect these changes.
+
+## View states
+
+<table class="temperatureViewStates">
+<tr><td><img src="images/temperatureControl.PNG" alt="Live state"/></td>
+    <td><img src="images/temperatureControl_edit.PNG" alt="Edit state"/></td></tr>
+<tr align="center"><td>Live State</td><td>Edit State</td></tr>
+</table>
+
+The **temperatureControlView** has 2 states - live & edit.
+
+In the live state, the live temperature is displayed but when the user left-clicks anywhere inside the thermometer or left-click on the mercury and drag it, the color of the mercury changes to blue (representing the edit state). Once in the edit state, user can change the value inside the textbox to change the mercury level too.
+
+To return back to the live state, the user can right-click anywhere inside the control.
+
+## API reference
+
+### Temperature Control Model
+
+**temperatureControlModel** extends L.Evented class and hence provides convenient subscription methods like *on(...)* & *off(...)*.
+
+#### Options
+
+| Property | Type | Description |
+|---|---|---|
+| value | number | The initial temperature value. It represents a temperature value in K. |
```

Comment:
Nope. K is internal representation. User of this class is only interested in the meaning of `value` property. If you write that it is a temperature in K then User may make a mistake and manually convert `value` according to `temperatureUnit` while `value` is already converted => invalid result.

Consider
```
var temperatureModel = ...
...
// let's display the temperature
var temperature = temperatureModel.value; 
// hmm, I'm confused what are the units here
// Let me check the documentation.
// Looks like it is in K. 
// Sadly but I'll have to do the manual conversion then.

var actualTemperature = convertFromK(temperatureModel.value, temperatureModel.temperatureUnit);
```